### PR TITLE
adds/fixes nested HAVING statements

### PR DESCRIPTION
**Pull request by cleathers**
This PR fixes nested HAVING statements 

I ran the `npm run dev` command but it updated more files than I expected it too. Is this expected?

### CODE REVIEW

File: lib/dialects/oracle/index.js [hunk 1]
```diff
@@ -74,8 +74,8 @@ assign(Client_Oracle.prototype, {
     var client = this;
     return new Promise(function (resolver, rejecter) {
       client.driver.connect(client.connectionSettings, function (err, connection) {
-        if (err) return rejecter(err);
         Promise.promisifyAll(connection);
+        if (err) return rejecter(err);
         if (client.connectionSettings.prefetchRowCount) {
```
Comment (cleathers) [inline lib/dialects/oracle/index.js]: https://github.com/tgriesser/knex/pull/896/files

this change wasn't made in src.

**Comment by cleathers** [comment]
Is there a way I can run the build process on just the files I've changed? There seems to be quite a bit of stuff getting reverted in this PR.

**Comment by rhys-vdw** [comment]
Hi @cleathers, I made a mistake merging #896. I'm not sure what else is being overridden, but just write the change as you intend and I'll merge it after master has been corrected.

**Comment by cleathers** [comment]
Sounds good @rhys-vdw. Thanks for your work on the library, it's a pretty awesome tool :)

File: src/query/builder.js [hunk 2]
```diff
@@ -481,16 +482,26 @@ assign(Builder.prototype, {
...
-      return this._havingRaw(column);
-    }
-    
     // Check if the column is a function, in which case it's
     // a having statement wrapped in parens.
     if (typeof column === 'function') {
       return this.havingWrapped(column);
     }
-    
+		  // Support "having true || having false"
```
Comment (elhigu) [inline src/query/builder.js]: Indentations are wrong here

File: src/query/compiler.js [hunk 3]
```diff
@@ -298,6 +276,51 @@ assign(QueryCompiler.prototype, {
     return this.update();
   },
 
+
+  // Having Clause
+  // ------
+
+  having: function() {
+		var havings = this.grouped.having;
```
Comment (elhigu) [inline src/query/compiler.js]: And indentations are wrong here too...

**Comment by elhigu** [comment]
Hi, I started going through old pull requests and get them merged if possible or close otherwise if they are not going to be finished.

@cleathers  Nice feature, thanks for this pull request. To be able to merge this, there were some strange indentation which needs to be fixed and documentation of the feature should be added / updated to `index.html`.

**Comment by elhigu** [comment]
Closing abandoned pull request and moved to issues for now https://github.com/tgriesser/knex/issues/1282

**Pull request closed without merging**